### PR TITLE
Added skin support

### PR DESCRIPTION
**Pull request by LPTP1Dev**
I made prismarine-viewer render player skins using their username

### CODE REVIEW

**Comment by rom1504** [comment]
can you run `npm run fix` to fix the lint?

**Comment by LPTP1Dev** [comment]
I'm about to change everything to backend so it's one request instead of two, after that of course

**Comment by rom1504** [comment]
What backend?

**Comment by LPTP1Dev** [comment]
lib/common.js

**Comment by rom1504** [comment]
Ah I see why not

**Comment by LPTP1Dev** [comment]
Now the "XHTMLRequest is not defined" errors should be gone too

File: lib/common.js [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+const fs = require('fs')
 
 function setupRoutes (app, prefix = '') {
   app.use(compression())
   app.use(prefix + '/', express.static(path.join(__dirname, '../public')))
+  app.get('/getSkin', (req, res) => {
+    if (!req.query.name) return res.status(400).send('Missing url')
+    const name = req.query.name
+
+    fetch('https://api.ashcon.app/mojang/v2/user/' + name)
```
Comment (Karang) [inline lib/common.js:14]: Why use a 3rd party api instead of the mojang one ? https://sessionserver.mojang.com/session/minecraft/profile/[uuid]

Comment (LPTP1Dev) [reply lib/common.js:14]: I started using this one cause a friend of mine recommended it to me, but if you want I can try to use the mojang apis instead

Comment (extremeheat) [reply lib/common.js:14]: mojang API will be more reliable and will look identical to how Minecraft client renders the skins, without any caching from 3rd party sites

```js
let uuid = await fetch(`https://api.mojang.com/users/profiles/minecraft/${username}?at=Date.now()`).then(r => r.json()).then(r => r.id)
let r = await fetch(`https://sessionserver.mojang.com/session/minecraft/profile/${uuid}`).then(r => r.json())
let skinData = JSON.parse(Buffer.from(r.properties[0].value, 'base64').toString('utf8'))
let url = skinData.textures.SKIN.url
```

**Comment by rom1504** [comment]
hi @LPTP1Dev do you want to finish this PR ?

**Comment by LPTP1Dev** [comment]
I will continue soon, have to do other Things rn

**Comment by rom1504** [comment]
@LPTP1Dev does it work ? should we merge ?

**Comment by LPTP1Dev** [comment]
It works, but as asked above, it does not use the official mojang servers. I'll try to get it running with the mojang services and push an update in the next days.

**Comment by LPTP1Dev** [comment]
There seems to be some problem with the CI workflow (the npm install) since everything is working fine for me and I did nothing with the gl module.

File: viewer/lib/entity/Entity.js [hunk 2]
```diff
@@ -200,12 +200,91 @@ function getMesh (texture, jsonModel) {
...
+  const geoData = {
+    positions: [],
+    normals: [],
+    uvs: [],
+    indices: [],
+    skinIndices: [],
+    skinWeights: []
+  }
+  let i = 0
+  jsonModel.default.bones.forEach(async jsonBone => {
```
Comment (extremeheat) [inline viewer/lib/entity/Entity.js:215]: Why the async lambda function?

**Review by extremeheat — APPROVED** [review]
LGTM setting aside unnecessary async

File: viewer/lib/entity/Entity.js (hunk 2 above)
Comment (LPTP1Dev) [reply viewer/lib/entity/Entity.js:215]: I am used to async cause I use await a lot and always forget to add the async, but I see there is no use here, let me quickly remove it here...

**Comment by rom1504** [comment]
failing and old

**Comment by LPTP1Dev** [comment]
> failing and old

I think you merged an [PR fixing the CI failing](https://github.com/PrismarineJS/prismarine-viewer/pull/346) some weeks ago.

**Comment by rom1504** [comment]
I did but this PR still fails after that.
Feel free to re open if you want to finish this

**Comment by LPTP1Dev** [comment]
Alright, see [here](https://github.com/PrismarineJS/prismarine-viewer/pull/360)

**Pull request closed without merging**